ce/ce/artifacts/activation.ts

```diff
@@ -474,101 +480,22 @@ export class Activation {
     return parts[n].split(delimiter).filter(each => each).map(each => `${front}${each}${back}`);
   }
 
-  async generateMSBuild(artifacts: Iterable<Artifact>): Promise<string> {
-    const msbuildFile = {
-      Project: {
-        $xmlns: 'http://schemas.microsoft.com/developer/msbuild/2003',
-        PropertyGroup: <Array<Record<string, any>>>[]
-      }
-    };
-
-    if (this.locationCount) {
-      const locations = <Record<string, any>>{
-        $Label: 'Locations'
-      };
-      for await (const [name, location] of this.locations) {
-        locations[name] = location;
-      }
-      msbuildFile.Project.PropertyGroup.push(locations);
-    }
-
-    if (this.propertyCount) {
-      const properties = <Record<string, any>>{
-        $Label: 'Properties'
-      };
-
-      for await (const [name, propertyValues] of this.properties) {
-        properties[name] = linq.join(propertyValues, ';');
-      }
-      msbuildFile.Project.PropertyGroup.push(properties);
-    }
-
-    if (this.toolCount) {
-      const tools = <Record<string, any>>{
-        $Label: 'Tools'
-      };
-
-      for await (const [name, tool] of this.tools) {
-        tools[name] = tool;
-      }
-      msbuildFile.Project.PropertyGroup.push(tools);
-    }
-
-    if (this.environmentVariableCount) {
-      const environment = <Record<string, any>>{
-        $Label: 'Environment'
-      };
-
-      for await (const [name, envValues] of this.environmentVariables) {
-        environment[name] = linq.join(envValues, ';');
+  generateMSBuild(): string {
```

Comment:
So is the intent now that everyone must duplicate _everything_ into MSBuild properties? For example: include paths, library link paths, defines, etc?

Reply:
I told Olga that there needs to be something reasonable done to INCLUDE et al., and she said that there is something that there needs to be done but she doesn't want to go there for the specific fact finding stuff we're doing vis a vis MSVC at this time.

Reply:
I am still thinking how the best to handle include paths, etc, The LIB, INCLUDE and PATH are redefined (and env vars reset) using project properties, so defining them in the generated .props does not do any good. And as the PATH is not inherited, it also overrides what is set for the PATH globally on the machine (which is bad).
I am leaning towards using new msbuild properties for the artifacts that don't define msbuil-properties but define include, lib and path and use them appropriately in the build process. But if an artifact defines msbuild-properties, I'd just use them (and nothing else) in the generated .props (at least by default).